plugins/woocommerce-blocks/assets/js/blocks/mini-cart/quantity-badge/style.scss

```diff
@@ -25,8 +25,8 @@
 
 :where(.wc-block-mini-cart__badge) {
 	// These values will be overridden in JS.
-	background-color: transparent;
-	color: transparent;
+	background-color: #111;
+	color: #f9f9f9;
```

Comment:
What is the purpose of these changes?

Reply:
With the transparent property, the color of the quantity badge is not visible. I have attached a video for an explanation.

https://github.com/woocommerce/woocommerce/assets/77401999/ddc1d05f-95b3-475b-b53c-c25acd2599e0

Reply:
Thanks, I see the issue now!

I wonder if we could undo this change, and add something like this in the `edit.tsx` file

 (similar logic is used in the mini-cart contents edit.tsx file)

```js
/**
 * This is a workaround for the Site Editor to set the correct
 * background color of the Mini-Cart QuantityBadge block based on
 * the main background color set by the theme.
 */
useEffect( () => {
	const canvasEl = document.querySelector(
		'.edit-site-visual-editor__editor-canvas'
	);
	if ( ! ( canvasEl instanceof HTMLIFrameElement ) ) {
		return;
	}
	const canvas =
		canvasEl.contentDocument || canvasEl.contentWindow?.document;
	if ( ! canvas ) {
		return;
	}
	if (
		canvas.getElementById( 'mini-cart-quantity-badge-foreground-color' )
	) {
		return;
	}
	const backgroundColor = window.getComputedStyle(
		canvas.body
	)?.backgroundColor;
	const color = window.getComputedStyle( canvas.body )?.color;

	if ( ! backgroundColor || ! color ) {
		return;
	}
	const style = document.createElement( 'style' );
	style.id = 'mini-cart-quantity-badge-foreground-color';

	// The badge's background should be the theme's foreground color, and it's text color the theme's background color.
	style.appendChild(
		document.createTextNode(
			`:where(.wc-block-mini-cart__badge) {
			color: ${ backgroundColor };
			background-color: ${ color };
		}`
		)
	);
	const body = canvas.querySelector( '.editor-styles-wrapper' );
	if ( ! body ) {
		return;
	}
	body.appendChild( style );
}, [] );
```

Reply:
Thanks for the wonderful suggestion 🙇. I have added the effect similar like this and also tested it by changing the theme bg color and it's working fine.